apps/src/templates/studioHomepages/JoinSection.jsx

```diff
@@ -62,21 +63,26 @@ const styles = {
   },
   clear: {
     clear: 'both'
+  },
+  dialog: {
+    padding: '20px'
   }
 };
 
 const INITIAL_STATE = {
-  sectionCode: ''
+  sectionCode: '',
+  displayCaptchaDialog: false
 };
 
 export default class JoinSection extends React.Component {
   static propTypes = {
     enrolledInASection: PropTypes.bool.isRequired,
     updateSections: PropTypes.func.isRequired,
-    updateSectionsResult: PropTypes.func.isRequired
+    updateSectionsResult: PropTypes.func.isRequired,
+    requireCaptcha: PropTypes.bool.isRequired
   };
 
-  state = {...INITIAL_STATE};
+  state = {...INITIAL_STATE, requireCaptcha: this.props.requireCaptcha};
```

Comment:
It's best practice to avoid copying props into state. https://reactjs.org/blog/2018/06/07/you-probably-dont-need-derived-state.html#anti-pattern-unconditionally-copying-props-to-state
Can you just use `displayCaptchaDialog` instead?